scripts/s3uploading.js

```diff
@@ -0,0 +1,116 @@
+const shell = require("shelljs");
+const AWS = require('aws-sdk');
+const fs = require('fs');
+const path = require('path');
+const crypto = require('crypto');
+
+const workdir = process.env.GITHUB_WORKSPACE;
+const s3 = new AWS.S3({
+    "AWS_ACCESS_KEY_ID": process.env.AWS_ACCESS_KEY_ID,
+    "AWS_SECRET_ACCESS_KEY": process.env.AWS_SECRET_ACCESS_KEY
+});
+const tmp = `${workdir}/tmp`;
+const readmeRegex = new RegExp('(readme\.md|readme\.org)', 'i');
+const Bucket = process.env.BUCKET_NAME
+
+const readJSON = (file) => JSON.parse(fs.readFileSync(file, 'utf8'));
+
+const modules = readJSON(`${workdir}/index.json`).modules;
+const versions = readJSON(`${workdir}/versions.json`);
+
+let commitMsg = ["Updated versions.json \n"];
+
+const uploadFile = (file, s3path) => s3.upload({
+    Bucket,
+    Key: s3path,
+    Body: fs.readFileSync(file),
+    ACL: 'public-read'
+}, (err, data) => {
+    if (err) console.error("Error while uploading", err);
+    if (data) console.log(`${data.Location} successfully uploaded`);
+})
+
+const createTMP = () => {
+    shell.rm('-rf', tmp);
+    shell.mkdir(tmp);
+}
+createTMP();
+
+const checkout = (module) => {
+    shell.exec(`git clone ${module.repo}`)
+    shell.exec(`git checkout ${module.commit}`)
```

Comment:
What if the commit doesn't exist?

Reply:
I still think there should be checks that the above commands succeeded.

Reply:
@vpodzime done, please re-review. Added checks, created an uploading queue which starts after all modules are checked out and archived. also cloning every module into individual directory